### ISSUE TITLE
OpenSSL NameError

### ISSUE THREAD

**Issue by ktk** [ruby-rdf/rdf-tabular#5]
I tried the command in #4 to map a CSV file and I get:

```
/usr/local/bundle/gems/rdf-2.2.3/lib/rdf/util/file.rb:109:in `open_url': uninitialized constant RDF::Util::File::NetHttpAdapter::OpenSSL (NameError)
Did you mean?  OpenStruct
...
```

Is there some dependency missing somewhere?

Install via gem, ruby 2.4

**Comment by gkellogg** [comment]
OpenSSL should be built in, count this be your problem? https://github.com/rvm/rvm/issues/3923

**Comment by ktk** [comment]
Hmm not sure, I tried the same in the official ruby containers for 2.2, 2.3 and 2.4 @ Dockerhub. Always exactly the same error.

**Comment by gkellogg** [comment]
Looks like it's missing a `require "openssl"` clause; I'll add this and push another update later today.

**Comment by ktk** [comment]
Great tnx

**Comment by ktk** [comment]
I tried the new version, good news is the SSL error is gone but I still can't get the example working:

```
[...]
,http://www.w3.org/ns/earl#TestSubject,,,,,,,http://www.w3.org/TR/tabular-metadata/,http://dbpedia.org/resource/Ruby_(programming_language),,,,,,,,,,,,,
,http://www.w3.org/ns/earl#Software,,,,,,,http://www.w3.org/TR/csv2rdf/,,,,,,,,,,,,,,
,,,,,,,,http://www.w3.org/TR/csv2json/,,,,,,,,,,,,,,'
	from /usr/local/lib/ruby/2.2.0/json/common.rb:155:in `parse'
	from /usr/local/bundle/gems/multi_json-1.12.1/lib/multi_json/adapters/json_common.rb:14:in `load'
	from /usr/local/bundle/gems/multi_json-1.12.1/lib/multi_json/adapter.rb:21:in `load'
	from /usr/local/bundle/gems/multi_json-1.12.1/lib/multi_json.rb:122:in `load'
	from /usr/local/bundle/gems/json-ld-2.1.2/lib/json/ld/api.rb:115:in `initialize'
	from /usr/local/bundle/gems/json-ld-2.1.2/lib/json/ld/api.rb:168:in `new'
	from /usr/local/bundle/gems/json-ld-2.1.2/lib/json/ld/api.rb:168:in `expand'
	from /usr/local/bundle/gems/json-ld-2.1.2/lib/json/ld/api.rb:409:in `toRdf'
	from /usr/local/bundle/gems/json-ld-2.1.2/lib/json/ld/reader.rb:61:in `each_statement'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/mixin/writable.rb:138:in `insert_statements'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/mixin/writable.rb:96:in `insert_reader'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/mixin/writable.rb:28:in `<<'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/mixin/mutable.rb:71:in `<<'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/cli.rb:370:in `block in exec'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/cli.rb:457:in `block (2 levels) in parse'
	from /usr/local/bundle/gems/json-ld-2.1.2/lib/json/ld/reader.rb:51:in `call'
	from /usr/local/bundle/gems/json-ld-2.1.2/lib/json/ld/reader.rb:51:in `block in initialize'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/reader.rb:286:in `instance_eval'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/reader.rb:286:in `initialize'
	from /usr/local/bundle/gems/json-ld-2.1.2/lib/json/ld/reader.rb:37:in `initialize'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/reader.rb:215:in `new'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/reader.rb:215:in `block in open'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/util/file.rb:331:in `open_file'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/reader.rb:194:in `open'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/cli.rb:455:in `block in parse'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/cli.rb:454:in `each'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/cli.rb:454:in `parse'
	from /usr/local/bundle/gems/rdf-2.2.5/lib/rdf/cli.rb:369:in `exec'
	from /usr/local/bundle/gems/rdf-2.2.5/bin/rdf:30:in `<top (required)>'
	from /usr/local/bundle/bin/rdf:22:in `load'
	from /usr/local/bundle/bin/rdf:22:in `<main>'

```

**Comment by gkellogg** [comment]
Yeah, the error reporting could use some work :smile:. But, in this case, it seems you're trying to parse your CSV file as JSON metadata. Perhaps you can provide a simple script which reproduces the problem. It will be a couple of days before I can look at it.

**Comment by ktk** [comment]
Basically all I tried is to convert the samples you gave me in #4:

    rdf serialize https://raw.githubusercontent.com/ruby-rdf/rdf-tabular/develop/etc/doap.csv --output-format ttl

If I run that out of the box I get that I don't have a writer for TTL and indeed it doesn't list it. If I change to ntriples, I get the error message I pasted above. The "validate" command from #4 results in a similar error, while the last example with `--input-format tabular` results in `rdf: No reader found for tabular.`

So basically I can't get anything of the examples to work.

**Comment by gkellogg** [comment]
You can remove the `--output-format ttl` to get as N-Triples. To get Turtle (TTL), install the  `linkeddata` gems.

The `tabular` format should show, even without `linkeddata` being installed, and I'll need to investigate that.

I'll add a comment to the README to suggest installing the `linkeddata` gem.

**Comment by gkellogg** [comment]
The `linkeddata` gem is necessary for the `tabular` format to work as the `rdf` command depends on that to discover other than a select number of different gems.

**Comment by ktk** [comment]
@gkellogg installing `linkeddata` gem apparently solved all issues, thanks!
